### PR TITLE
Create :jnt reader conditional support

### LINKED ISSUE #68: Add a jaunt reader conditional
I'm not sure what the precise semantics should be here, but it could be good to add a Reader Conditional, say `#clojarr` which people can use to maintain libraries if Clojarr diverges from Clojure, and the package maintainers want to help both stay compatible.

### PR DESCRIPTION
**Pull request by arrdem**
This changeset has one primary change: supporting the `:jnt` conditional clause in addition to `:clj`.

``` clojure
#?(:cljs "nope" :jnt "w00t" :clj "nope")
;; => "w00t"

#?(:cljs "nope" :clj "still works")
;; => "still works"
```

Incidental changes to this end include:
- Upgrading to Java 1.8
- Providing an interface default implementation of `clojure.lang.Seqable/iterator` (Java 1.8 feature)
- Adding `clojure.lang.RT/union [set, coll] -> set` (used in the reader changes)

Fixes #68

### CODE REVIEW

**Comment by arrdem** [comment]
Needs reader tests...

File: src/jvm/clojure/lang/Seqable.java [hunk 1]
```diff
@@ -12,6 +12,12 @@
 
 package clojure.lang;
 
-public interface Seqable {
+import java.util.Iterator;
+
+public interface Seqable extends Iterable {
   ISeq seq();
+
+default Iterator iterator() {
```
Comment (andrewhr) [inline src/jvm/clojure/lang/Seqable.java:20]: :wine_glass:

Comment (arrdem) [inline src/jvm/clojure/lang/Seqable.java:20]: :beer:

**Comment by arrdem** [comment]
:shipit:

**Pull request merged**